src/util/psbt/macros.rs

```diff
@@ -12,12 +12,17 @@
 // If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 //
 
+// #[allow(unused_macros)]
+// macro_rules! hex_psbt {
+//     ($s:expr) => { $crate::consensus::deserialize::<$crate::util::psbt::PartiallySignedTransaction>(&<$crate::prelude::Vec<u8> as $crate::hashes::hex::FromHex>::from_hex($s).unwrap()) };
+// }
+
```

Comment:
Just a personal tip: it makes review easier if the diff is clean (doesn't have stuff like this in it), anything you can do to make review easier helps you keep reviewers happy which helps you get your stuff reviewed.

Reply:
Thank you for excusing the mess. Won't happen again.

Reply:
> I reverted the hex_psbt macro changes too and just changed the test serialize_and_deserialize_global to use the new ser/deser code.

Shouldn't the hex_psbt macro serialize any psbt from hex? After fixing off by 1, tests still fail for e.g. InvalidMagic. Doesn't that mean there's an issue with the deserialze method? At first I thought the macro may deserialize any psbt-related piece of data with the consensus-encoding, but it looks like it's only deserializing Psbts. And this code behaves in ways the tests do not expect.

```
---- util::psbt::tests::bip_371_vectors::valid_psbt_vectors stdout ----
thread 'util::psbt::tests::bip_371_vectors::valid_psbt_vectors' panicked at 'called `Result::unwrap()` on an `Err` value: Psbt(InvalidMagic)', src/util/psbt/mod.rs:897:626
```

Reply:
Looks like the cause of the failing tests is serializing with `Encodable` and trying to deserialize with `Deserialize`.

Reply:
Doesn't that mean one of the two is non-standard? Bip 174 defines a byte sequence on the wire